scripts/prod/zero-downtime-deploy.sh

```
@@ -0,0 +1,231 @@
+    fi
+}
+
+is_port_in_use() {
+    local port=$1
+    lsof -t -i:$port > /dev/null 2>&1
+    return $?
+}
+
+kill_process_on_port() {
+    local port=$1
+    local pid=$(lsof -t -i:$port 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)